### PR TITLE
Rem 676

### PR DESCRIPTION
**Pull request by Alinadimmokdad**
iv-ava page 
new page open list form contain table with S/L columns contains button open serial form or lot form

### CODE REVIEW

File: src/pages/iv-ava/form/AvailabilityList.js [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+import CustomTextField from 'src/components/Inputs/CustomTextField'
+import { InventoryRepository } from 'src/repositories/InventoryRepository'
+import SerialForm from './SerialForm'
+import LotForm from './LotForm'
+
+const AvailabilityList = ({ itemId, sku, name, qty, labels, access }) => {
+  const { getRequest } = useContext(RequestsContext)
+  const { stack } = useWindow()
+
+  const [data, setData] = useState([])
```
Comment (imtithalhalalo) [inline src/pages/iv-ava/form/AvailabilityList.js:21]: why do you need useState here cannot use useResourceQuery?

File: src/pages/iv-ava/form/AvailabilityList.js [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+  async function fetchGridData(options = {}) {
+    const { _startAt = 0, _pageSize = 50 } = options
+
+    const response = await getRequest({
+      extension: InventoryRepository.Availability.qry,
+      parameters: `_itemId=${itemId}&_siteId=${0}&_startAt=${_startAt}&_pageSize=${_pageSize}&filter=`
+    })
+
+    const filteredData = response.list.filter(item => item.siteRef && item.siteName)
```
Comment (imtithalhalalo) [inline src/pages/iv-ava/form/AvailabilityList.js:31]: add null check` response?.list?.filter(`

File: src/pages/iv-ava/form/AvailabilityList.js [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+          )
+        }
+
+        return null
+      }
+    }
+  ]
+
+  const onSerial = obj => {
+    openSerialForm(obj.itemId, obj.siteId)
```
Comment (imtithalhalalo) [inline src/pages/iv-ava/form/AvailabilityList.js:80]: you can pass obj and then use it inside the form

File: src/pages/iv-ava/form/AvailabilityList.js [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+  }
+
+  const onLot = obj => {
+    openLotForm(obj.lotCategoryId, obj.itemId, obj.siteId)
```
Comment (imtithalhalalo) [inline src/pages/iv-ava/form/AvailabilityList.js:84]: same for here

File: src/pages/iv-ava/form/LotForm.js [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+        field: 'udt2',
+        headerName: response.record.udt2,
+        flex: 1
+      })
+    }
+
+    setColumns(dynamicColumns)
+  }
+  useEffect(() => {
+    getDynamicColumns(), fetchGridData()
```
Comment (imtithalhalalo) [inline src/pages/iv-ava/form/LotForm.js:95]:
```
getDynamicColumns()
fetchGridData()
```

File: src/pages/iv-ava/index.js [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+      field: 'netPrice',
+      headerName: labels.netPrice,
+      flex: 1
+    }
+  ]
+
+  const edit = obj => {
+    openAvailability(obj.itemId, obj.qty, obj.sku, obj.name)
+    console.log(obj)
+  }
```
Comment (imtithalhalalo) [inline src/pages/iv-ava/index.js:102]: pass just obj and then use and remove console.log

**Comment by imtithalhalalo** [comment]
Screenshot 2024-11-01 111306
page not added

**Comment by FatimaSalami** [comment]
image
why do we have the x icon when here we can't delete the field's content or edit it

**Review by FatimaSalami — APPROVED** [review]

**Pull request merged**